app/src/main/kotlin/io/intrepid/fablekotlin/screens/login/LoginPresenter.kt

```diff
@@ -0,0 +1,29 @@
+package io.intrepid.fablekotlin.screens.login
+
+import io.intrepid.fablekotlin.base.BasePresenter
+import io.intrepid.fablekotlin.base.PresenterConfiguration
+import io.intrepid.fablekotlin.models.LogInUserRequest
+import io.intrepid.fablekotlin.models.LogInUserResponse
+
+class LoginPresenter(view: LoginContract.View, configuration: PresenterConfiguration)
+    : BasePresenter<LoginContract.View>(view, configuration), LoginContract.Presenter {
+
+    override fun getAndStoreUserToken(tempToken: String) {
+
+        val logInUserObservable = restApi.logInUser(LogInUserRequest(tempToken))
+
+        val disposable = logInUserObservable
+                .subscribeOnIoObserveOnUi()
+                .subscribe({ logInUserResponse: LogInUserResponse ->
+                        userSettings.token = (logInUserResponse.authentication.authToken!!)
```

Comment:
Something to look at is if theres a way to get here without using the "!!", I dont know enough about Kotlin to know if we can make this happen without it but that would prevent getting any null errors here.

Reply:
I know using ?? adds a sorta built in null check like "do the thing if it's not null, if it's null skip over it" but I had null checks here originally but they became unneccesary apparently with some of my other code already doing null checks with ?? but I have to double check what that was and if that actually applies. I just remember it telling me the null checks were always true.

Reply:
edit: if you saw my previous comment ignore the part about I had null checks earlier I was thinking of a different part of the code after looking.

Reply:
edit 2.0: ignore what I just said there WERE null checks I'm just blind

Reply:
alright I put back in the null checks from my original code to get what it was saying and I was originally checking whether logInUserResponse was null and if logInUserResponse.authentication was null but kotlin claims they will always be true